Add info on trigger id's uniqueness requirement (#19336)
home-assistant/home-assistant.github.io: source/_docs/automation/trigger.markdown

## Trigger id

> All triggers can be assigned an optional `id`. If the id is omitted, it will instead be set to the index of the trigger. The `id` can be referenced from trigger conditions. The `id` needs to be unique for each trigger (i.e. you can not use the `id` field to group triggers).

Reviewer: ```suggestion
All triggers can be assigned an optional `id`. If the id is omitted, it will instead be set to the index of the trigger. The `id` can be referenced from trigger conditions. The `id` should be unique for each trigger.
```
It _can_ be the same, but shouldn't be.
Grouping is not a context we have here, but in other places so mentioning it here will add more confusion.
